testing/infra/terraform/modules/ec_deployment/variables.tf

```
@@ -24,10 +24,10 @@ variable "stack_version" {
   type        = string
 }
 
-variable "monitor_deployment" {
-  default     = false
-  type        = bool
-  description = "Optionally monitor the deployment in a separate deployment"
+variable "observability_deployment" {
+  default     = "none"
+  type        = string
+  description = "Optional deployment ID, for platform observability. Use 'self' to send data to the same deployment, 'dedicated' to create a dedicated observability deployment, or 'none' (the default) to disable observability. Any other value is assumed to be the ID of an existing, externally managed Elasticsearch deployment."
```

Can we add a validation rule for this variable? Since the values are fixed (or deployment ID which is of length 32), and are expected to be one of a list, it fits the `validation` use case quite nicely: https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules.

```suggestion
  description = "Optional deployment ID, for platform observability. Use 'self' to send data to the same deployment, 'dedicated' to create a dedicated observability deployment, or 'none' (the default) to disable observability. Any other value is assumed to be the ID of an existing, externally managed Elasticsearch deployment."
  validation {
    condition     = can(regex("^self$|^dedicated$|^none$", var.observability_deployment)) || length(var.observability_deployment) == 32
    error_message = "invalid observability_deployment value. Must be one of 'self','dedicated','none', '<deployment id>'"
  }
```

---

Thanks for the tip, will do.